### DISCUSSION TITLE
Solved - Help Need - Using media buttons

### DISCUSSION THREAD

**Discussion by SuperNinja-4965** [hathach/tinyusb#620] [Q&A] [answered]
I am not sure if this is the right place to be asking this but here goes!
I am using the tinyusb library with my Raspberry pi pico. i have a button connected to the pico. When the button is pressed a subroutine is called which uses `tud_hid_keyboard_report` to press a key on the keyboard.
All is working fine, I can do key combinations type sentances etc.

But what i cannot work out is how to use/ press media keys. I would like my button to say for example pause my music when i press it. (I am using the hid_composite example with modifications to use my button)
Is there a way to do this. I am new to TinyUSB and so i have probably missed something here.

When i search through the TinyUSB Repo i find `HID_USAGE_CONSUMER_PLAY_PAUSE` so i assume it is possible somehow.

**Comment by SuperNinja-4965** [comment]
You can probably guess this but i am coding the pico in c/c++

**Comment by hathach** [comment]
For media control, you need to use the  `Consumer Control` hid profile, I will update the hid_composite example to also send the volumn down key for demonstration. For environment, it shouldn't be issue with either pico sdk or not, I prefer to compile and run the stock example here, since it is my normal routine to test cross-platform example on multiple mcus.

**Reply by SuperNinja-4965** [reply]
thanks very much!

**Comment by hathach** [comment, marked as answer]
@joshuag1000 check out the modification in the hid_composite example. https://github.com/hathach/tinyusb/pull/622 , you could quickly test it out without compiling by downloading the rp2040 artifacts file from the ci build https://github.com/hathach/tinyusb/actions/runs/527018529

**Reply by SuperNinja-4965** [reply]
Ok so i have merged your code with mine and that volume key down example works brilliantly! I think i understand how the code works better now! 
Thanks!

**Reply by SuperNinja-4965** [reply]
@hathach ok i dont know if this is an issue with my code or no but when i change the key to either or the next track or previous track buttons spotify goes crazy and starts skipping tracks in that direction with no end until i unplug the device.

**Reply by SuperNinja-4965** [reply]
actually i think that is because i am using `HID_USAGE_CONSUMER_SCAN_NEXT`. Which i am guessing is supposed to do that?? Is there a way i can have it so i only skip one track?

**Reply by SuperNinja-4965** [reply]
Turns out there was a little instability in my code causing that. Its all patched up!
TYSM for the help!
